text/3488-deprecate-cargo-categories.md

```diff
@@ -0,0 +1,374 @@
+
+
+# Keywords vs. Categories
+
+**Keywords** (aka. tags) describe an open set of names to categorize packages.
+The set being "open" means that users can use arbitrary keywords without having
+to choose from a fixed list of available names.
+
+**Categories** describe a closed set of names to categorize packages. The set 
+being "closed" means that users have to choose from a fixed list of available
+categories, and can't easily add new ones.
+
+**Comparison:**
+
+- **Creation:** As described above, the keywords concept makes it
+quite easy to create new names by simply using them in the `Cargo.toml` file.
+With categories, a pull request to the crates.io repository has to be opened,
+followed by a discussion among the crates.io team on whether there might be
+enough justification for adding such a new category.
+
+- **Typos:** The chance of misspelling categories is reduced since cargo and
+crates.io can warn about misspelled category names. Keywords can be arbitrary,
+so a typo-detection is not quite as easy.
+
+  - It would be possible to compare keywords against a list of commonly used
+  keywords though and warn the user about potential typos.
+
+- **Duplicates:** Since the list of categories is fixed and new additions are
+checked by the crates.io team, the chance for having categories with similar
+purposes is small. With keywords, it is easy to have both `math` and
+`mathematics` keywords available on crates.io. In that case it would be the
+burden of the crate authors to use the most appropriate and/or commonly used
+keywords.
+
+  - This could be countered to some degree by having a list of common aliases
+  on the server-side that are applied either at publish-time or when a user
+  searches for a particular keyword. [Experience from libs.rs indicates that
+  there is no trivial solution to this though](https://github.com/rust-lang/rfcs/pull/3488#issuecomment-1721208225) 
+
+- **Clarity:** Categories, as implemented on crates.io, have a description
+associated with them, so the chance of misunderstanding what kinds of crates
+belong to them is reduced. Keywords by themselves can be unclear in some
+cases. One example being the [`cli` keyword](https://github.com/rust-lang/crates.io/pull/488#issuecomment-265600770)
+which includes libraries for building CLI applications, as well as CLI
+applications themselves.
+
+- **Politics:** There are users feeling that having a category exist on
+crates.io can be seen as an endorsement. This makes the introduction
+and deletion of categories feel political to some degree. One example of this
+being a [discussion about the "cryptocurrencies" category](https://github.com/rust-lang/crates.io/discussions/6762),
+and whether it should be deleted or not. Keywords do not have this issue as
+the crates.io team has no direct control over the creation of keywords (except
+for ones that violate e.g. the [Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct)).
```

Comment:
This is mostly being pedantic, but I prefer to avoid the term "political" when referring to controversy since it tends to make most folks shut off the critical thinking faculties in their brains that elaborate on why things are controversial. So, here's a suggestion for a way to word this without using the term:

```suggestion
- **Authority:** Some users feel that having a category exist on crates.io is an
endorsement of that category by the Rust project. This makes the introduction
and deletion of categories hold more weight than they should, and it puts an
increased burden of discussion for all those involved. One example of this
being a [discussion about the "cryptocurrencies" category](https://github.com/rust-lang/crates.io/discussions/6762), and
whether it should be deleted or not. Keywords don't have this issue since they
aren't added with any involvement from the crates.io team, and thus don't have
the same authority. The crates.io team can still remove keywords that violate 
e.g. the [Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),
but that's much easier to decide than having to take a stance on whether
keywords are endorsed or not.
```

Reply:
Sounds great, though I would replace "Authority" with "Endorsement" to match the phrasing in the text: 

```suggestion
- **Endorsement:** Some users feel that having a category exist on crates.io is an
endorsement of that category by the Rust project. This makes the introduction
and deletion of categories hold more weight than they should, and it puts an
increased burden of discussion for all those involved. One example of this
being a [discussion about the "cryptocurrencies" category](https://github.com/rust-lang/crates.io/discussions/6762), and
whether it should be deleted or not. Keywords don't have this issue since they
aren't added with any involvement from the crates.io team, and thus don't have
the same authority. The crates.io team can still remove keywords that violate 
e.g. the [Code of Conduct](https://www.rust-lang.org/policies/code-of-conduct),
but that's much easier to decide than having to take a stance on whether
keywords are endorsed or not.
```